Voycar.Api.Web/Generic/Repository/Repository.cs

```diff
@@ -53,7 +53,11 @@ private bool SaveChanges()
     /// </returns>
     public bool Update(TEntity entity)
     {
-        this.DbSet.Update(entity);
+        var existingEntity = this.DbSet.Find(entity.Id);
+        if (existingEntity != null)
+        {
+            this.DbSet.Entry(existingEntity).CurrentValues.SetValues(entity);
+        }
```

Comment:
Interessant. Und das macht jetzt, was wir wollen?

Reply:
Ja, das Problem beim anderen Update war, dass einfach ein neues Entity in die Datenbank eingefügt wurde. Wenn ich allerdings dem Entity die entsprechende ID mitgegeben habe, dann kam ein Fehler, weil die ID schon vorhanden ist. So macht es was es soll, aber ich dachte eig. , dass das Update das auch so macht. Anscheinend ja nicht.